core/color.cpp

```diff
@@ -268,100 +253,75 @@ Color Color::from_rgbe9995(uint32_t p_rgbe) {
 	return Color(rd, gd, bd, 1.0f);
 }
 
-static float _parse_col(const String &p_str, int p_ofs) {
-
-	int ig = 0;
+static int _parse_col(const String &p_str, int p_ofs) {
+	int col_comp = 0;
 
 	for (int i = 0; i < 2; i++) {
-
-		int c = p_str[i + p_ofs];
-		int v = 0;
+		int c = p_str[p_ofs + i];
 
 		if (c >= '0' && c <= '9') {
-			v = c - '0';
+			col_comp |= c - '0';
 		} else if (c >= 'a' && c <= 'f') {
-			v = c - 'a';
-			v += 10;
+			col_comp |= 10 + c - 'a';
 		} else if (c >= 'A' && c <= 'F') {
-			v = c - 'A';
-			v += 10;
+			col_comp |= 10 + c - 'A';
 		} else {
 			return -1;
 		}
 
 		if (i == 0)
-			ig += v * 16;
-		else
-			ig += v;
+			col_comp <<= 4;
```

Comment:
Is this bit-shift safe to perform twice in a row? The variable is stored outside of the loop.

This code could probably be improved further - it's less readable compared to the rest of the code. The old code for this method is also confusing.

Reply:
`col_comp` is set to `0` at the beginning, so all its bits are zero.
The first iteration of the for loop will set the last 4 bits of `col_comp`, and the shift will move these 4 bits so the second iteration doesn't overwrite them. The second iteration then writes the next 4 bits, so the bit shift only runs once.

The function takes a component of a hex color code e.g. `FF` or `2A`, and converts it to an int (more specifically the last 8 bits of the int), which is value from 0 - 255.
Changing the function name may help make it more descriptive; possibly adding a comment as well.

I am aware that this bit of code requires knowledge of text encoding and bit operations to understand what the function is doing, but I think it is required to change a hex color code as a string to an 8-bit color code.